packages/next/client/image.tsx

```diff
@@ -362,555 +363,579 @@ function handleLoading(
+      config,
+      src,
+      unoptimized,
+      width: widthInt,
+      quality: qualityInt,
+      sizes,
+      loader,
+    })
 
-  let srcString: string = src
+    let srcString: string = src
 
-  if (process.env.NODE_ENV !== 'production') {
-    if (typeof window !== 'undefined') {
-      let fullUrl: URL
-      try {
-        fullUrl = new URL(imgAttributes.src)
-      } catch (e) {
-        fullUrl = new URL(imgAttributes.src, window.location.href)
+    if (process.env.NODE_ENV !== 'production') {
+      if (typeof window !== 'undefined') {
+        let fullUrl: URL
+        try {
+          fullUrl = new URL(imgAttributes.src)
+        } catch (e) {
+          fullUrl = new URL(imgAttributes.src, window.location.href)
+        }
+        allImgs.set(fullUrl.href, { src, priority, placeholder })
       }
-      allImgs.set(fullUrl.href, { src, priority, placeholder })
     }
-  }
 
-  const linkProps: React.DetailedHTMLProps<
-    React.LinkHTMLAttributes<HTMLLinkElement>,
-    HTMLLinkElement
-  > = {
-    // @ts-expect-error upgrade react types to react 18
-    imageSrcSet: imgAttributes.srcSet,
-    imageSizes: imgAttributes.sizes,
-    crossOrigin: rest.crossOrigin,
-  }
+    const linkProps: React.DetailedHTMLProps<
+      React.LinkHTMLAttributes<HTMLLinkElement>,
+      HTMLLinkElement
+    > = {
+      // @ts-expect-error upgrade react types to react 18
+      imageSrcSet: imgAttributes.srcSet,
+      imageSizes: imgAttributes.sizes,
+      crossOrigin: rest.crossOrigin,
+    }
 
-  const onLoadRef = useRef(onLoad)
-
-  useEffect(() => {
-    onLoadRef.current = onLoad
-  }, [onLoad])
-
-  const onLoadingCompleteRef = useRef(onLoadingComplete)
-
-  useEffect(() => {
-    onLoadingCompleteRef.current = onLoadingComplete
-  }, [onLoadingComplete])
-
-  const imgElementArgs: ImageElementProps = {
-    isLazy,
-    imgAttributes,
-    heightInt,
-    widthInt,
-    qualityInt,
-    className,
-    imgStyle,
-    blurStyle,
-    loading,
-    config,
-    fill,
-    unoptimized,
-    placeholder,
-    loader,
-    srcString,
-    onLoadRef,
-    onLoadingCompleteRef,
-    setBlurComplete,
-    setShowAltText,
-    ...rest,
+    const onLoadRef = useRef(onLoad)
+
+    useEffect(() => {
+      onLoadRef.current = onLoad
+    }, [onLoad])
+
+    const onLoadingCompleteRef = useRef(onLoadingComplete)
+
+    useEffect(() => {
+      onLoadingCompleteRef.current = onLoadingComplete
+    }, [onLoadingComplete])
+
+    const imgElementArgs: ImageElementProps = {
+      isLazy,
+      imgAttributes,
+      heightInt,
+      widthInt,
+      qualityInt,
+      className,
+      imgStyle,
+      blurStyle,
+      loading,
+      config,
+      fill,
+      unoptimized,
+      placeholder,
+      loader,
+      srcString,
+      onLoadRef,
+      onLoadingCompleteRef,
+      setBlurComplete,
+      setShowAltText,
+      ...rest,
+    }
+    return (
+      <>
+        {<ImageElement {...imgElementArgs} ref={forwardedRef} />}
```

Comment:
Should we add this to `imgElementArgs` instead?

Reply:
That would make typing more verbose. I would need to `omit<>` it from the type. Because `React.forwardRef` API takes those types separately.